sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -2610,8 +2610,11 @@ case class SubtractDates(
 
   override def inputTypes: Seq[AbstractDataType] = Seq(DateType, DateType)
   override def dataType: DataType = {
-    // TODO(SPARK-35727): Return INTERVAL DAY from dates subtraction
-    if (legacyInterval) CalendarIntervalType else DayTimeIntervalType()
+    if (legacyInterval) {
+      CalendarIntervalType
+    } else {
+      DayTimeIntervalType(DayTimeIntervalType.DAY, DayTimeIntervalType.DAY)
```

Comment:
I have added `apply()` with single arg recently https://github.com/apache/spark/pull/32997, you can use it here as:
```scala
DayTimeIntervalType(DAY)
```